### ISSUE TITLE
[DoesNotReturn] warning is only reported in nullable-enabled context

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#56706]
Warning CS8763 should be reported for `F1()` below:
```csharp
using System.Diagnostics.CodeAnalysis;

class Program
{
    [DoesNotReturn] static void F1() { } // no warning

#nullable enable
    [DoesNotReturn] static void F2() { } // warning CS8763
}

namespace System.Diagnostics.CodeAnalysis
{
    public class DoesNotReturnAttribute : Attribute { }
}
```

**Comment by RikkiGibson** [comment]
I think the behavior in the issue description is expected. DoesNotReturn is not respected by control flow analysis, definite assignment, etc. Is there some design notes/documentation indicating otherwise?
